File: 2020/python/day6.py
with open('../day_6.txt', 'r') as f:
    answers = [[set(person) for person in group.split()] for group in f.read().split('\n\n')]

p1 = sum(len(set.union(*group)) for group in answers)
p2 = sum(len(set.intersection(*group)) for group in answers)

print('part 1:', p1)
print('part 2:', p2)

File: 2021/ruby/day6.rb
fish = [0] * 9
File.read("../day_6.txt").split(",").map(&:to_i).each { |f| fish[f.to_i] += 1 }

def simulate(n, fish)
  n.times do
    fish.rotate!
    fish[6] += fish[8]
  end
  fish.sum
end

print("part 1: ", simulate(80, fish.dup), "\n")
print("part 2: ", simulate(256, fish.dup), "\n")
